source/plugin/permissions.rst

```diff
@@ -2,9 +2,293 @@
+obtained from the ``PermissionService``. It is recommended that plugins define their own permissions to the default's
+transient ``SubjectData`` during every server start-up. This allows the server owner to overwrite the defaults defined
+by plugins according to his needs using the default's persistent ``SubjectData``.
+
+.. note::
+
+    Its only specified for the default ``Subjects`` that the persistent ``SubjectData`` takes precedence over transient
+    ones.
+
+If neither the Subject, nor any of its parents, nor the defaults grant and deny a permission, then it is automatically
+denied.
+
+.. note::
+
+    Order of precedence in descending order:
+    
+    * Subject itself
+    * Parent Subjects
+    * SubjectCollection Defaults
+        * Persistent
+        * Transient
+    * PermissionService Defaults
+        * Persistent
+        * Transient
+    * Deny permission
+
+SubjectCollections
+==================
+
+A container for subjects that can be used to obtain a Subject by name.
+These are the default Subject Collections:
+
+* User
+    * Contains all on-line ``Players``. It may also return subjects for off-line ``Users``. However there aren't any
+      specifications regarding this, so they might be fully transient dummy subjects.
+* Group 
+    * Contains all group ``Subject``. Groups are a simple way of structuring a ``Subject's`` inheritance tree.
+* System
+    * Contains other ``Subjects`` used by the server such as the the console and possible remote consoles. 
+* Command Block
+    * Contains all ``Subjects`` for command blocks. These are useful if you would like to run a ``CommandBlock`` only
+      with the permissions of the creator.
+* Role Template
+    * Contains all role template subjects that are used in ``PermissionDescriptions``. Useful to lookup all recommended
+      permissions for a user. These should not be used for inheritance.
+
+SubjectData
+===========
+
+SubjectData are the actual permission stores connected to the Subject.
+There are two types of Subject stores:
+
+* Transient = Only lasts for the duration of the session, it is never saved
+* Regular (persistent) = Might be saved somewhere, and therefore be persisted and exist forever. Its recommended for
+  ``PermissionServices`` to implement a persistent store, however it is not a requirement. It might also depend on the
+  subject type. If there is no persistence then the transient store will be returned in both methods.
+
+Plugin authors should consider whether it is necessary to persist a value when choosing between them.
+
+* If it is only for a short time (ex during a minigame) then use the transient one.
+* If it is for a long time or forever (ex a promotion to VIP) use the regular (persistent) one.
+
+Except for the default's subject data it is not specified which one takes precedence over the other.
+It is not recommended to set it to both stores at the same time to overcome the unspecified precedence.
```

Comment:
I think it was agreed among current plugin authors that for defaults only, having persistent take priority makes the most sense. IMO, this should therefore become a requirement within the API. It's not in the JDs yet, so idk if we want to exclude this section from SpongeDocs until it gets PRed into SpongeAPI.

I'm not sure if zml has made this change for PEX yet, but as far as I know, he plans to.